### PR TITLE
Runner Discovered Suites

### PR DESCRIPTION
**Pull request by cheeseng**
Expose discovered suites through Runner's discoveredSuites.

### CODE REVIEW

**Comment by bvenners** [comment]
@neko-kai Can you test out this PR to see if it adequately addresses your needs?

**Comment by neko-kai** [comment]
@bvenners Yes, I just tested – it fully solves the problem! (https://github.com/7mind/izumi/pull/2120)

**Comment by cheeseng** [comment]
@bvenners I resolved the conflicts, I can merge this in after your review.

Thanks.

File: native/core/src/main/scala/org/scalatest/tools/MasterRunner.scala [hunk 1]
```diff
@@ -164,11 +166,18 @@ class MasterRunner(theArgs: Array[String], theRemoteArgs: Array[String], testCla
...
+    Runner.setDiscoveredSuites(discoveredSuites)  
+
     def createTask(t: TaskDef): Task =
       new TaskRunner(t, testClassLoader, tracker, tagsToInclude, tagsToExclude, t.selectors() ++ autoSelectors, false, presentAllDurations, presentInColor, presentShortStackTraces, presentFullStackTraces, presentUnformatted, presentReminder,
         presentReminderWithShortStackTraces, presentReminderWithFullStackTraces, presentReminderWithoutCanceledTests, presentFilePathname, presentJson, Some(send))
 
-    (if (wildcard.isEmpty && membersOnly.isEmpty) taskDefs else (filterWildcard(wildcard, taskDefs) ++ filterMembersOnly(membersOnly, taskDefs)).distinct).map(createTask)
+    val tasks = (if (wildcard.isEmpty && membersOnly.isEmpty) taskDefs else (filterWildcard(wildcard, taskDefs) ++ filterMembersOnly(membersOnly, taskDefs)).distinct).map(createTask)
+
+    //runnerInstance().internalDiscoveredSuites.set(Some(discoveredSuites))
```
Comment (bvenners) [inline native/core/src/main/scala/org/scalatest/tools/MasterRunner.scala:178]: @cheeseng Why is this line of code commented out?

**Review by bvenners — APPROVED** [review]

File: native/core/src/main/scala/org/scalatest/tools/MasterRunner.scala (hunk 1 above)
Comment (cheeseng) [reply native/core/src/main/scala/org/scalatest/tools/MasterRunner.scala:178]: @bvenners The discovered suites are set in line 170 already, I think I just forgot to remove this commented line, this feature is supported on native as well.

**Pull request merged**